geoai/train.py

```
@@ -1860,11 +1870,22 @@ def train_MaskRCNN_model(
             f"Epoch {epoch+1}/{num_epochs}: Train Loss: {train_loss:.4f}, Val Loss: {eval_metrics['loss']:.4f}, Val IoU: {eval_metrics['IoU']:.4f}"
         )
 
-        # Save best model
-        if eval_metrics["IoU"] > best_iou:
-            best_iou = eval_metrics["IoU"]
+        # Checking IoU improvement for earlystop and prevent overfitting
+        if current_iou > best_iou + early_stopping_min_delta:
+            best_iou = current_iou
+            epochs_without_improvement = 0
             logger.info(f"Saving best model with IoU: {best_iou:.4f}")
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)